File: compiler/bench/mutrec2.c
// expected: 10000000
#include <stdio.h>
long g2(long n);
long f2(long n) {
    if (n == 0) return 0;
    return g2(n - 1) + 1;
}
long g2(long n) {
    if (n == 0) return 0;
    return f2(n - 1) + 1;
}
int main() {
    long sum = 0;
    for (long i = 0; i < 100000; i++) {
        sum += f2(100);
    }
    printf("%ld\n", sum);
    return 0;
}

File: compiler/bench/mutrec2.fs
\ expected: 10000000
\ 2 mutually recursive functions, 10M total calls
\ f calls g, g calls f, alternating down to 0

: f2 ( n -- result ) recursive
  dup 0= if exit then
  1- g2 1+ ;

: g2 ( n -- result ) recursive
  dup 0= if exit then
  1- f2 1+ ;

: main
  0 100000 0 do
    100 f2 +
  loop . cr ;
